lib/queue.rb

```diff
@@ -1,31 +1,68 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @size = 10
+    @store = Array.new(@size)
+    @front = 0
+    @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # Check for full
+   if @front == (@back + 1) % @size
+      self.expand
+    end
```

Comment:
This defeats the circular nature of the list, so that you can work with a fixed size, but it does let you resize the buffer.